packages/create-cloudflare/src/frameworks/nuxt/index.ts

```diff
@@ -29,7 +31,8 @@ const config: FrameworkConfig = {
 	displayName: "Nuxt",
 	packageScripts: {
 		"pages:dev": `wrangler pages dev ${compatDateFlag()} --proxy 3000 -- npm run dev`,
-		"pages:deploy": `NODE_VERSION=17 npm run generate && wrangler pages deploy ./dist`,
+		"pages:deploy":
+			"NITRO_PRESET=cloudflare-pages npm run build && wrangler pages deploy ./dist",
```

Comment:
Should the `npm run build` command ever be run without the `NITRO_PRESET=cloudflare-pages` env var? Should we move the env var into the build command?

Reply:
mh.... yes that's a totally valid point... 🤔 

You shouldn't run `npm run build` without the env var
but I don't think there is a huge benefit in that since:
 - if you deploy via `pages:deploy` you do get the env var
 - `pages:dev` doesn't use `npm run build` so for local development there is no difference
 - if you want to deploy via the Pages CI you get the env var from the environment there already
 
 One issue here is that `pagesScripts` are used to add/override scripts in package.json:
https://github.com/cloudflare/workers-sdk/blob/113fb7fe86cde0c43a7efa7fc2e58ec8c55d8267/packages/create-cloudflare/src/pages.ts#L117
so I would either have to:
- refactor the `packageScripts` logic to somehow allow modification of scripts instead of overriding them (in this case prepending) 
- override the `build` script generated by the Nuxt CLI, which seems less robust than the current solution

🤔 
....

Actually the more I think of this the more refactoring the `packageScripts` seems like the most proper solution to me... what do you think?

@jculvey do you have an opinion on this? (and if you say we should update the logic, any preference on the api?)

Reply:
I'm a little hesitant to refactor because I can't think of many more use cases in c3 right now that would benefit from modifying commands, so I'm leaning towards just keeping `NITRO_PRESET` in the deploy command.

If we were to refactor it I think the API of `packageScripts` would change to take a function instead of a string, where the function receives the existing command. Something like: 

```
	packageScripts: {
		"pages:build": (cmd) => `NITRO_PRESET ${cmd}`,
    ...
  }
```

Without another use case though it's hard to tell if that enhancement would be enough to accommodate any future use cases we might encounter.

Reply:
yeah that improvement looks like a good idea 🙂

anyways as I mentioned it might be not too important right now....

maybe we could go with the current implementation and if later people complain about it we just chance course?

I totally don't mind adding the above refactoring so that we still allow `packageScripts` to be strings but optionally if one is a function then we run it and use its output, it actually doesn't sound too difficult/problematic anyways...

I don't mind, whatever you guys prefer 🙂 (please make the final call 😁)

Reply:
`I totally don't mind adding the above refactoring so that we still allow packageScripts to be strings but optionally if one is a function then we run it and use its output, it actually doesn't sound too difficult/problematic anyways...`

Go for it!

Reply:
Cool 🙂 , I've converted the PR to draft just to make sure we don't merge it until I implement that 🙂

Reply:
Updated, please have a look @jculvey 🙏